homeassistant/components/apcupsd/config_flow.py

```diff
@@ -0,0 +1,161 @@
+"""Config flow for APCUPSd integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.config_entries import (
+    CONN_CLASS_LOCAL_POLL,
+    SOURCE_IMPORT,
+    ConfigEntry,
+    ConfigFlow,
+    OptionsFlow,
+)
+from homeassistant.const import CONF_HOST, CONF_PORT, CONF_RESOURCES
+from homeassistant.core import callback
+from homeassistant.data_entry_flow import FlowResult
+from . import DOMAIN, APCUPSdData
+from .sensor import SENSOR_TYPES
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class ConfigFlowHandler(ConfigFlow, domain=DOMAIN):
+    """APCUPSd integration config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = CONN_CLASS_LOCAL_POLL
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
+        """Get the options flow for this handler."""
+        return OptionsFlowHandler(config_entry)
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            # Test if connection to the host is ok and get the current status for later
+            # configuration.
+            data_service = APCUPSdData(user_input[CONF_HOST], user_input[CONF_PORT])
+            try:
+                status: dict[str, Any] | None = await self.hass.async_add_executor_job(
+                    lambda: data_service.status
+                )
+                if status is None:
```

Comment:
Please only wrap the line that can raise inside the try... except block. Use an `else:` block if needed.

Reply:
Done (this part is rewritten)